Template instanciation error : lager/detail/signal.hpp

I encounter the following errors when attempting to compile this simple example from the doc.
https://gist.github.com/odecaux/17cc74b005dbbdf76948ccaf153b93b7

I'm using the latest commit :  d0359a3

I have the exact same ouptut on both clang-cl 10.0.0, and clang 10.0.1 on my linux machine
```
In file included from F:\Sources\Lager\First_Example\main.cpp:6:
In file included from F:\Sources\vcpkg\installed\x86-windows\include\lager/store.hpp:17:
In file included from F:\Sources\vcpkg\installed\x86-windows\include\lager/state.hpp:16:
In file included from F:\Sources\vcpkg\installed\x86-windows\include\lager/cursor.hpp:16:
In file included from F:\Sources\vcpkg\installed\x86-windows\include\lager/detail/nodes.hpp:41:
F:\Sources\vcpkg\installed\x86-windows\include\lager/detail/signal.hpp(49,58): error: too few arguments to function call, expected 2, have 1
        void operator()(Args... args) final { fn_(args...); }
                                              ~~~        ^
F:\Sources\vcpkg\installed\x86-windows\include\lager/detail/signal.hpp(46,9): note: in instantiation of member function 'lager::detail::signal<const model &>::slot<void (*)(model, model)>::operator()' requested here
        slot(Fn fn)
        ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\memory(2833,32): note: in instantiation of member function 'lager::detail::signal<const model &>::slot<void (*)(model, model)>::slot' requested here
    return unique_ptr<_Ty>(new _Ty(_STD forward<_Types>(_Args)...));
                               ^
F:\Sources\vcpkg\installed\x86-windows\include\lager/detail/signal.hpp(66,29): note: in instantiation of function template specialization 'std::make_unique<lager::detail::signal<const model &>::slot<void (*)(model, model)>, void (&)(model, model), 0>' requested here
        auto s       = std::make_unique<slot_t>(std::forward<Fn>(fn));
                            ^
F:\Sources\vcpkg\installed\x86-windows\include\lager/watch.hpp(89,34): note: in instantiation of function template specialization 'lager::detail::signal<const model &>::connect<void (&)(model, model)>' requested here
        conns_.push_back(base_t::connect(std::forward<CallbackT>(callback)));
                                 ^
F:\Sources\vcpkg\installed\x86-windows\include\lager/watch.hpp(99,18): note: in instantiation of function template specialization 'lager::watchable_base<lager::detail::store_node_base<std::variant<increment_action, decrement_action, reset_action>, model> >::watch<void (&)(model, model)>' requested here
    return value.watch(std::forward<CallbackT>(callback));
                 ^
1 error generated.
```

gcc 9.3.0 (linux machine) gives me : 
```
In file included from /home/odecaux/libraries/lager/lager/detail/nodes.hpp:41,
                 from /home/odecaux/libraries/lager/lager/cursor.hpp:16,
                 from /home/odecaux/libraries/lager/lager/state.hpp:16,
                 from /home/odecaux/libraries/lager/lager/store.hpp:17,
                 from main.cpp:6:
/home/odecaux/libraries/lager/lager/detail/signal.hpp: In instantiation of ‘void lager::detail::signal<Args>::slot<Fn>::operator()(Args ...) [with Fn = void (*)(model, model); Args = {const model&}]’:
/home/odecaux/libraries/lager/lager/detail/signal.hpp:49:14:   required from here
/home/odecaux/libraries/lager/lager/detail/signal.hpp:49:47: error: too few arguments to function
   49 |         void operator()(Args... args) final { fn_(args...); }
        |                                               ^~~
```

--- a/doc/architecture.rst
+++ b/doc/architecture.rst
@@ -279,16 +279,25 @@ purposes now:
 
 .. code-block:: c++
 
-    void draw(counter::model prev, counter::model curr)
+    void draw(counter::model curr)
     {
-        std::cout << "last value: " << prev.value << '\n';
         std::cout << "current value: " << curr.value << '\n';
     }
 
-.. tip:: The view function has access to both the last and new value
-         of the application.  This allows the view to decide which
-         parts of the view do need to be updated based on the
-         differences between the old and new models.
+.. tip:: In previous versions of the library, the view function used
+         be provided the previous value of the model too.  This
+         allowed the view to decide which parts of the view do need to
+         be updated based on the differences between the old and new
+         model values.  If this behavior is wanted, it can be
+         simulated by wrapping the view function in a lambda like
+         this.
+
+         .. code-block:: c++
+
+             [prev = store.get()] (auto curr) mutable {
+                 draw(prev, curr);
+                 prev = curr;
+             }
 
 
 Glueing things together

